### ISSUE TITLE
 Nancy.ConfigurationException:on Centos V7

### ISSUE THREAD

**Issue by escamoteur** [NancyFx/Nancy#2436]
### Description

I tried to run Nancy on a fresh installed Cento VM machine. I got this exception:

```
Unhandled Exception:
Nancy.ConfigurationException: Error loading default confguration for Nancy.GlobalizationConfiguration ---> System.TypeInitializationException: The type initializer for 'Nancy.GlobalizationConfiguration' threw an exception. ---> Nancy.ConfigurationException: Invalid Globalization configuration. You must support at least one culture
  at Nancy.GlobalizationConfiguration..ctor (IEnumerable`1 supportedCultureNames, System.String defaultCulture) <0x41307d30 + 0x001ef> in <filename unknown>:0
  at Nancy.GlobalizationConfiguration..cctor () <0x41307c60 + 0x0009f> in <filename unknown>:0
  --- End of inner exception stack trace ---
  at Nancy.Configuration.NancyDefaultConfigurationProvider`1[T].Nancy.Configuration.INancyDefaultConfigurationProvider.GetDefaultConfiguration () <0x41305960 + 0x00016> in <filename unknown>:0
  at Nancy.Configuration.DefaultNancyEnvironmentConfigurator.SafeGetDefaultConfiguration (INancyDefaultConfigurationProvider configurationProvider) <0x41305870 + 0x00029> in <filename unknown>:0
  --- End of inner exception stack trace ---
  at Nancy.Configuration.DefaultNancyEnvironmentConfigurator.SafeGetDefaultConfiguration (INancyDefaultConfigurationProvider configurationProvider) <0x41305870 + 0x000cf> in <filename unknown>:0
  at Nancy.Configuration.DefaultNancyEnvironmentConfigurator.ConfigureEnvironment (System.Action`1 configuration) <0x41305570 + 0x00087> in <filename unknown>:0
  at Nancy.Bootstrapper.NancyBootstrapperBase`1[TContainer].Initialise () <0x40a6ab30 + 0x00357> in <filename unknown>:0
  at Nancy.Hosting.Self.NancyHost..ctor (INancyBootstrapper bootstrapper, Nancy.Hosting.Self.HostConfiguration configuration, System.Uri[] baseUris) <0x40a6a970 + 0x0010d> in <filename unknown>:0
  at Nancy.Hosting.Self.NancyHost..ctor (System.Uri[] baseUris) <0x40a61950 + 0x00057> in <filename unknown>:0
  at QuoteMyDayServer.Program.Main (System.String[] args) <0x40a59d50 + 0x000a7> in <filename unknown>:0
[ERROR] FATAL UNHANDLED EXCEPTION: Nancy.ConfigurationException: Error loading default confguration for Nancy.GlobalizationConfiguration ---> System.TypeInitializationException: The type initializer for 'Nancy.GlobalizationConfiguration' threw an exception. ---> Nancy.ConfigurationException: Invalid Globalization configuration. You must support at least one culture
  at Nancy.GlobalizationConfiguration..ctor (IEnumerable`1 supportedCultureNames, System.String defaultCulture) <0x41307d30 + 0x001ef> in <filename unknown>:0
  at Nancy.GlobalizationConfiguration..cctor () <0x41307c60 + 0x0009f> in <filename unknown>:0
  --- End of inner exception stack trace ---
  at Nancy.Configuration.NancyDefaultConfigurationProvider`1[T].Nancy.Configuration.INancyDefaultConfigurationProvider.GetDefaultConfiguration () <0x41305960 + 0x00016> in <filename unknown>:0
  at Nancy.Configuration.DefaultNancyEnvironmentConfigurator.SafeGetDefaultConfiguration (INancyDefaultConfigurationProvider configurationProvider) <0x41305870 + 0x00029> in <filename unknown>:0
  --- End of inner exception stack trace ---
  at Nancy.Configuration.DefaultNancyEnvironmentConfigurator.SafeGetDefaultConfiguration (INancyDefaultConfigurationProvider configurationProvider) <0x41305870 + 0x000cf> in <filename unknown>:0
  at Nancy.Configuration.DefaultNancyEnvironmentConfigurator.ConfigureEnvironment (System.Action`1 configuration) <0x41305570 + 0x00087> in <filename unknown>:0
  at Nancy.Bootstrapper.NancyBootstrapperBase`1[TContainer].Initialise () <0x40a6ab30 + 0x00357> in <filename unknown>:0
  at Nancy.Hosting.Self.NancyHost..ctor (INancyBootstrapper bootstrapper, Nancy.Hosting.Self.HostConfiguration configuration, System.Uri[] baseUris) <0x40a6a970 + 0x0010d> in <filename unknown>:0
  at Nancy.Hosting.Self.NancyHost..ctor (System.Uri[] baseUris) <0x40a61950 + 0x00057> in <filename unknown>:0
  at QuoteMyDayServer.Program.Main (System.String[] args) <0x40a59d50 + 0x000a7> in <filename unknown>:0

```

The same App was running fine on Ubuntu server
### System Configuration
- Nancy version: V2
- Nancy host
  - [ ] ASP.NET
  - [ ] OWIN
  - [x] Self-Hosted
  - [ ] Other:
- Other Nancy packages and versions:
  - Nancy.Validation.DataAnnotations v2.0
  - Nancy.Authentication.Stateless
- Environment (Operating system, version and so on): CentOs v7
- .NET Framework version: v4.5

**Comment by jchannon** [comment]
It's because Mono sees a invariant culture for some reason.
@akoeplinger recommends setting the LANG / LC_ALL environment variables

**Comment by escamoteur** [comment]
The question is if it's the best solution to throw an exception instead to fall back to en_US

**Comment by jchannon** [comment]
I think it's a lower level issue TBH.

**Comment by khellang** [comment]
So, according to [the constructor](https://github.com/NancyFx/Nancy/blob/2c07700b2d999d20c60f524480c48b2dc19b678c/src/Nancy/GlobalizationConfiguration.cs#L34), the exception is thrown if all supported cultures passed in are `null` or empty. That means that Mono returns `null` or an empty string for `CultureInfo.CurrentCulture.Name`. Isn't that a bit weird? Doesn't invariant culture have a name?

**Comment by Jetski5822** [comment]
Argh just hit this on mono :(

**Comment by jchannon** [comment]
I think @escamoteur fixed it by setting the ENV VAR

**Comment by escamoteur** [comment]
I'm just trying to remeber how I fixed it because I had problems with the ENV VAR

**Comment by thecodejunkie** [comment]
Is this a "centos thing" or are we likely to run into this on multiple Mono + Linux setups?

**Comment by escamoteur** [comment]
Ah I know, I switched to Ubuntu server and there is was already set.

**Comment by jchannon** [comment]
This is more a Linux/Mono issue as different distros handle the languages
differently so the current thread culture/invariant culture differs

**Comment by thecodejunkie** [comment]
Can we put any fail-safe in place? Better error message?

**Comment by escamoteur** [comment]
This would be a great help at least, or fall back to default culture lile en_EN_UTF-8

**Comment by thecodejunkie** [comment]
I think a fallback culture can be bad and have very unexpected side effects depending on what you use the culture.

**Comment by escamoteur** [comment]
Can a culture be set in the BootStrapper?

**Comment by jchannon** [comment]
Yes in the Configure method

**Comment by thecodejunkie** [comment]
If this might be a common thing on Linux distros when we need to try and work around it in someway..  the least we should do it provide a better error message to let people know why they're seeing the error and how they can explicitly configure a supported culture

**Comment by thecodejunkie** [comment]
Imma going to bring in @yantrio into this

**Comment by ehajri** [comment]
Just got this error on Mono on Windows 7. fyi.

**Comment by jchannon** [comment]
Why use mono on Windows?

**Comment by ehajri** [comment]
Running my project on Mono/Mac OS X got so many issues.
So I figured, if I get the project to work on Mono on my Windows box, then having the same working on Mac won't be a big issue.

**Comment by escamoteur** [comment]
If the issues were not mono related, it makes no sense to use mono on windows. Get your project running under visual studio and then port to mono

**Comment by ehajri** [comment]
Thanks but yes they are :) I couldn't just plug and play.

**Comment by ehajri** [comment]
Okay I thought I'd share this.

Initially, my project, as a whole, had issues with Mono (System.Configuration, Microsoft.ReportViewer.RunTime, etc..) so I decided to go back to my windows box, resolve all issues, then port back to Mac OS X.

As all other issues were resolved, I ran into this exception. The comments from @jchannon and @escamoteur made me doubt my findings so I went ahead and tested the same using Mono on Mac.

The could got compiled, and executed perfectly. I did not even need to change culture using:
```c#
public override void Configure(INancyEnvironment environment)
{
    environment.Cultures(new[] { "en-US" }, "en-US");
}
```
It's all good (well, except it doesn't work on Mono/Windows but who really cares..)
